0029-loopback-transport/0029-loopback-transport.md

```diff
@@ -0,0 +1,33 @@
+---
+title: The Loopback Transport Protocol (LT)
+draft: 1
+---
+# Loopback Transport Protocol (LT)
+
+The Loopback Transport protocol is a technique for transporting value over the Interledger. It relies on the establishment of a temporary, direct, ILPv4 link between sender and receiver (the loopback link). The flow is as follows.
+
+## Flow
+
+The following flow describes how the sender's loopback transfer module ("LT module") interacts with the application layer protocol, the receiver, and the ILPv4 network.
+
+1. The loopback connection is established out of band. For instance, the application layer protocol may allow the receiver to specify a [BTP/2.0](../0023-bilateral-transfer-protocol/0023-bilateral-transfer-protocol.md) URL for the sender to connect to.
+1. The sender's LT module is initialized with this open connection.
+1. The sender's LT module discovers the loopback address using [IL-DCP](https://github.com/interledgerjs/ilp-protocol-ildcp).
+1. The sender's application layer protocol tells the LT module to send a payment for a given source amount, and expiry time.
+1. The sender's LT module chooses a fulfillment at random.
+1. The sender's LT module creates the condition of their payment by computing the SHA-256 hash of the fulfillment.
+1. The sender's LT module sets the destination address to the loopback address for this lookback link.
+1. Using this condition, address, and amount, the sender's LT module prepares an ILPv4 payment to a connector of her choice.
+1. The payment will find its way from this connector, via any number of additional connectors, to the receiver.
+1. The receiver checks that the destination address matches the loopback address he announced to the sender over IL-DCP earlier, and forwards the ILPv4 `prepare` message to the sender, over the loopback link.
```

Comment:
What if the receiver doesn't want it? How would they communicate that to the sender?

Reply:
that's part of a higher-level protocol; for this spec we assume Alice wants to pay Bob and Bob wants to be paid.

Reply:
Can that higher level protocol communicate over the channel established for loopback or does it need a separate channel?

Reply:
sure, as long as there's a way to distinguish the different types of messages

Reply:
Does this protocol give you a way to do that or if that also out of scope?

Reply:
I think what you're trying to say with this thread is that there are advanced features (like updating the requested amount of a payment that is already in progress) that are implemented in https://github.com/interledgerjs/ilp-protocol-stream (which by the way, is awesome! I really like what you're working on there!) that are not implemented in LT. And you're right, LT is a very simple transport protocol, that does not have such advanced features. Let's chat more about it at our face-to-face meeting next week!